core/domain/user_jobs_one_off.py

```diff
@@ -135,6 +139,97 @@ def reduce(key, stringified_values):
                 subscription_services.subscribe_to_collection(key, item['id'])
 
 
+class UserImpactCalculationOneOffJob(jobs.BaseMapReduceJobManager):
+    """ Calculates each user's impact score, where impact score is defined as:
+    Sum of (
+    ln(playthroughs) * (ratings_scaler) * (average(ratings) - 2.5))
+    *(multiplier),
+    where multiplier = 10, and ratings_scaler is .1(number of ratings)
+    if there are < 10 ratings for that exploration.
+
+    Impact scores are calculated over explorations for which a user
+    is listed as a contributor.
+    """
+
+    MULTIPLIER = 10
+    MIN_AVERAGE_RATING = 2.5
+    NUM_RATINGS_SCALER_CUTOFF = 10
+    NUM_RATINGS_SCALER = .1
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @classmethod
+    def _get_exp_impact_score(cls, exploration_id):
+        from core.domain import rating_services
```

Comment:
Try and put these at the top of the file if you can -- we're trying to remove inline imports from the codebase (see #1149). If you can't, and you get import errors, then it's fine to leave it as-is.